### PR TITLE
Add whitaker_clones_core: rustc_lexer norm, k-shingling, winnow (7.2.1)

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Introduced the whitaker_clones_core crate providing rustc_lexer-based token normalization, k-shingling, 64-bit Rabin-Karp hashing, and winnowing for clone detection.
- Wired the new crate into the workspace with a public API surface including normalize, hash_shingles, winnow, and token/types exports.
- Added unit and BDD tests, plus docs/execplan and roadmap updates to align with the 7.2.1 workstream.
- Public API surfaces are defined within the new crate; this stage focuses on core library functionality and test coverage.
- Status: 7.2.1 complete; next workstreams (7.2.2 MinHash, 7.2.3 SARIF/CLI) planned and scoped.

> Note: This work aligns with ExecPlan 7.2.1 and the ongoing clone-detector scaffolding.

## Changes
- Added crates/whitaker_clones_core with core modules and public API:
  - lib.rs exposing Normalize, hash_shingles, winnow, and token/types exports.
  - token/mod.rs, token/normalize.rs, token/fingerprint.rs, token/error.rs, token/types.rs, token/tests.rs.
- Implemented token normalization pipeline using rustc_lexer, plus per-token handling via a centralized path (process_token in the normalization module).
- Introduced domain types and validation in token/types.rs: NormProfile, ShingleSize, WinnowWindow, NormalizedToken, NormalizedTokenKind, IdentifierSymbol, LiteralSymbol, Fingerprint, and related helpers.
- Added unit tests and behaviour tests scaffolding for token normalization, shingling, hashing, and winnowing:
  - crates/whitaker_clones_core/src/token/tests.rs
  - crates/whitaker_clones_core/tests/token_pass_behaviour.rs
  - crates/whitaker_clones_core/tests/features/token_pass.feature
- Updated workspace Cargo.toml to include whitaker_clones_core and wire rustc_lexer as a workspace dependency.
- Documentation/assets added/refreshed for 7.2.1:
  - docs/execplans/7-2-1-rustc-lexer-normalization.md
  - docs/roadmap.md updated to reflect 7.2.1 completion status
  - docs/whitaker-clone-detector-design.md updated with 7.2.1 decisions
- Updated Task references and preserved existing test scaffolding to ensure determinism, error handling, and fingerprinting behavior.

## Why this matters
- Centralizes per-token logic into a cohesive, testable, and extensible pipeline.
- Establishes a stable foundation for subsequent 7.2.2/7.2.3 workstreams (MinHash, SARIF emission, CLI wiring) while keeping public APIs stable for consumers.
- Enables deterministic normalization semantics across T1 and T2 profiles, with clearly defined error paths for unterminated literals and unsupported tokens.

## Testing plan
- Unit tests cover T1/T2 normalization, shingling, rolling hashes, and winnowing edge cases (zero sizes, short inputs).
- Behaviour tests exercise end-to-end token normalization, fingerprint generation, and winnowing, including error paths.
- doctests/examples exercised via cargo test --doc for public API surface.
- Tests continue to validate determinism and fingerprinting semantics; unhappy paths are exercised through explicit TokenPassError variants.

## Documentation and roadmap
- ExecPlan 7.2.1 documentation updated to reflect the new token-engine crate, its responsibilities, and gating for 7.2.2/7.2.3.
- Roadmap entry 7.2.1 updated to complete status in docs/roadmap.md.
- Clone-detector design docs updated to capture final 7.2.1 decisions and crate boundaries.

## Acceptance criteria mapping
- New ExecPlan doc for 7.2.1 added and aligned with plan.
- Clone-detector core scaffolding created and wired to workspace dependencies.
- Design docs updated to reflect 7.2.1 decisions and gating strategy.
- Roadmap updated to reflect 7.2.1 completion status.
- No rustc_private dependencies introduced in this stage; public API surfaces are documented.

📎 Task: https://www.devboxer.com/task/024c96d5-10d4-4f5d-bb79-6a6d9a145bc6
</body>

📎 **Task**: https://www.devboxer.com/task/64625dbd-1657-4043-99ba-02ad2fb87732

### CODE REVIEW

File: crates/whitaker_clones_core/src/token/fingerprint.rs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    let mut last_index = None;
+
+    for start in 0..=(fingerprints.len() - width) {
+        let end = start + width;
+        let index = start + rightmost_minimum_index(&fingerprints[start..end]);
+        if last_index != Some(index) {
+            retained.push(fingerprints[index].clone());
+            last_index = Some(index);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/whitaker_clones_core/src/token/fingerprint.rs]: @coderabbitai Has this now been resolved in the latest commit?

Use codegraph analysis to determine your answer.

If this comment is now resolved, please mark it as such using the API. Otherwise, please provide an AI agent prompt for the remaining work to be done to address this comment.

File: crates/whitaker_clones_core/src/token/types.rs [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    }
+}
+
+/// Literal representation for `T1` and `T2`.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum LiteralSymbol {
+    /// Preserve the original literal source text.
+    Original(String),
+    /// Canonicalize the literal to a category marker.
+    Canonical(&'static str),
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/whitaker_clones_core/src/token/types.rs]: @coderabbitai "Canonicalize" is correct for en-GB-oxendict (-ize / -yse / -our)
